### PR TITLE
make sign_assertion and sign_response configurable

### PR DESCRIPTION
**Pull request by leifj**
The saml2int profile actually requires the Assertion to be signed in addition to the response. We might want to make this configurable maybe...? however the default should be saml2int right?

### CODE REVIEW

**Comment by leifj** [comment]
Yes you're right. I'll update the PR to support signed assertions but turn it off as default

**Comment by leifj** [comment]
I went "nuts" and made both assertion and response signing configurable but with different defaults. My thinking is that we may need to support crazy corner-cases. I've also clarified documentation which was inconsistent with code in the use of idp_config/sp_config

File: src/satosa/frontends/saml2.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import json
 import logging
 from urllib.parse import urlparse
+from distutils.util import strtobool
```
Comment (johanlundberg) [inline src/satosa/frontends/saml2.py]: strtobool should not be needed as the config is yaml and when loaded parses True/yes/true to boolean True and False/no/false to boolean False.

**Comment by skoranda** [comment]
The changes will not allow a per-SP configuration to be applied if there is no explicit default set because, for example, 

sign_assertion = self.config['idp_config']['service']['idp']['policy']['default']['sign_assertion']

will throw a KeyError and then the line

sign_assertion = self.config['idp_config']['service']['idp']['policy'][resp_args['sp_entity_id']]['sign_assertion']

will never have a chance to be evaluated.

Was that intended or should there be two (2) try/except clauses each for sign_assertion and sign_response?

**Comment by johanlundberg** [comment]
Damn, I should have spotted that. I will fix it.

**Pull request merged**